Index: Notification/src/main/java/com/example/Notification/model/NotificationTemplate.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/model/NotificationTemplate.java b/Notification/src/main/java/com/example/Notification/model/NotificationTemplate.java
new file mode 100644
--- /dev/null	(date 1746867121286)
+++ b/Notification/src/main/java/com/example/Notification/model/NotificationTemplate.java	(date 1746867121286)
@@ -0,0 +1,14 @@
+package com.example.Notification.model;
+
+import lombok.*;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Document(collection = "notification_templates")
+
+public class NotificationTemplate {
+}
Index: Notification/src/main/java/com/example/Notification/repository/NotificationRepository.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/repository/NotificationRepository.java b/Notification/src/main/java/com/example/Notification/repository/NotificationRepository.java
new file mode 100644
--- /dev/null	(date 1746866392795)
+++ b/Notification/src/main/java/com/example/Notification/repository/NotificationRepository.java	(date 1746866392795)
@@ -0,0 +1,5 @@
+package com.example.Notification.repository;
+
+public interface NotificationRepository {
+    // MongoRepo later
+}
Index: Notification/src/main/java/com/example/Notification/model/Notification.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/model/Notification.java b/Notification/src/main/java/com/example/Notification/model/Notification.java
new file mode 100644
--- /dev/null	(date 1746867031683)
+++ b/Notification/src/main/java/com/example/Notification/model/Notification.java	(date 1746867031683)
@@ -0,0 +1,24 @@
+package com.example.Notification.model;
+
+import lombok.*;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.time.LocalDateTime;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Document(collection = "notifications")
+public class Notification {
+    @Id
+    private String id;                      // MongoDB unique ID (automatically generated)
+
+    private Long userId;                    // Who receives the notification
+    private Long bookingId;                 // Related booking
+
+    private NotificationType type;          // EMAIL or SMS
+    private String message;                 // Final rendered message (with template filled in)
+    private LocalDateTime timestamp;        // When it was created
+}
Index: Notification/src/main/java/com/example/Notification/service/NotificationService.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/service/NotificationService.java b/Notification/src/main/java/com/example/Notification/service/NotificationService.java
new file mode 100644
--- /dev/null	(date 1746866508705)
+++ b/Notification/src/main/java/com/example/Notification/service/NotificationService.java	(date 1746866508705)
@@ -0,0 +1,5 @@
+package com.example.Notification.service;
+
+public interface NotificationService {
+    // define service contract
+}
Index: Notification/src/main/java/com/example/Notification/controller/NotificationController.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/controller/NotificationController.java b/Notification/src/main/java/com/example/Notification/controller/NotificationController.java
new file mode 100644
--- /dev/null	(date 1746866454873)
+++ b/Notification/src/main/java/com/example/Notification/controller/NotificationController.java	(date 1746866454873)
@@ -0,0 +1,8 @@
+package com.example.Notification.controller;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/notifications")
+public class NotificationController {
+// endpoints later
+}
Index: Notification/src/main/java/com/example/Notification/listener/BookingNotificationListener.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/listener/BookingNotificationListener.java b/Notification/src/main/java/com/example/Notification/listener/BookingNotificationListener.java
new file mode 100644
--- /dev/null	(date 1746866566785)
+++ b/Notification/src/main/java/com/example/Notification/listener/BookingNotificationListener.java	(date 1746866566785)
@@ -0,0 +1,6 @@
+package com.example.Notification.listener;
+
+public class BookingNotificationListener {
+    // @RabbitListener will go here
+
+}
Index: Notification/src/main/resources/application.yml
===================================================================
diff --git a/Notification/src/main/resources/application.yml b/Notification/src/main/resources/application.yml
new file mode 100644
--- /dev/null	(date 1746865802542)
+++ b/Notification/src/main/resources/application.yml	(date 1746865802542)
@@ -0,0 +1,12 @@
+server:
+  port: 8083
+
+spring:
+  application:
+    name: notification-service
+  data:
+    mongodb:
+      uri: mongodb://localhost:27017/notifications
+  rabbitmq:
+    host: localhost
+    port: 5672
Index: Notification/src/main/java/com/example/Notification/strategy/NotificationSender.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/strategy/NotificationSender.java b/Notification/src/main/java/com/example/Notification/strategy/NotificationSender.java
new file mode 100644
--- /dev/null	(date 1746866681094)
+++ b/Notification/src/main/java/com/example/Notification/strategy/NotificationSender.java	(date 1746866681094)
@@ -0,0 +1,5 @@
+package com.example.Notification.strategy;
+
+public interface NotificationSender {
+    void send(String to, String message);
+}
Index: Notification/src/main/java/com/example/Notification/NotificationServiceApplication.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/NotificationServiceApplication.java b/Notification/src/main/java/com/example/Notification/NotificationServiceApplication.java
new file mode 100644
--- /dev/null	(date 1746865748270)
+++ b/Notification/src/main/java/com/example/Notification/NotificationServiceApplication.java	(date 1746865748270)
@@ -0,0 +1,11 @@
+package com.example.Notification;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class NotificationServiceApplication {
+    public static void main(String[] args) {
+        SpringApplication.run(NotificationServiceApplication.class, args);
+    }
+}
Index: Notification/src/main/java/com/example/Notification/dto/BookingNotificationEvent.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/dto/BookingNotificationEvent.java b/Notification/src/main/java/com/example/Notification/dto/BookingNotificationEvent.java
new file mode 100644
--- /dev/null	(date 1746866747460)
+++ b/Notification/src/main/java/com/example/Notification/dto/BookingNotificationEvent.java	(date 1746866747460)
@@ -0,0 +1,5 @@
+package com.example.Notification.dto;
+
+public class BookingNotificationEvent {
+    // DTO fields from RabbitMQ
+}
Index: Notification/src/main/java/com/example/Notification/template/NotificationTemplateProcessor.java
===================================================================
diff --git a/Notification/src/main/java/com/example/Notification/template/NotificationTemplateProcessor.java b/Notification/src/main/java/com/example/Notification/template/NotificationTemplateProcessor.java
new file mode 100644
--- /dev/null	(date 1746866719073)
+++ b/Notification/src/main/java/com/example/Notification/template/NotificationTemplateProcessor.java	(date 1746866719073)
@@ -0,0 +1,5 @@
+package com.example.Notification.template;
+
+public abstract class NotificationTemplateProcessor {
+    // template processor logic
+}
